pkg/storage-init/storage-init.sh

```diff
@@ -155,7 +157,22 @@ for BLK_DEVICE in $BLK_DEVICES; do
 done
 
 #Recording SMART details to a file
-smartctl -a "$(grep -m 1 /persist < /proc/mounts | cut -d ' ' -f 1)" --json > $SMART_DETAILS_FILE
+SMART_JSON=$(smartctl -a "$(grep -m 1 /persist < /proc/mounts | cut -d ' ' -f 1)" --json)
+CURRENT_POWER_CYCLE_COUNT=$(echo "$SMART_JSON" | grep -o '\bpower_cycle_count[^,]*' | cut -d ':' -f 2 | sed -e 's/^[[:space:]]*//')
```

Comment:
What does this generate when the device does not support smartctl?

Reply:
Can't nodeagent do this processing? Seems less error prone than doing it in the script.

Reply:
Moved the processing of previous and current SMART data to nodeagent.